docs: update Module Federation guide (#6746)
web-infra-dev/rspack: website/docs/en/guide/features/module-federation.mdx

## Use case

Module Federation has some typical use cases, including:

- Allow independent applications (known as "microfrontends" in microfrontend architecture) to share modules without the need to recompile the entire application.
- Distributed teams to work on different parts of the same application without the need to recompile the entire application.
- Dynamic code loading and sharing between applications during runtime

Module Federation can help you:

- Reduce code duplication
- Improve code maintainability
- Lower the overall size of your applications
- Enhance the performance of your applications

## How to use

There are currently several versions of Module Federation, you can choose one to use:

### Module Federation v1.0

This version was implemented for compatibility with the [webpack.container.ModuleFederationPlugin](https://webpack.js.org/plugins/module-federation-plugin/).

You can use it through Rspack's [ModuleFederationPluginV1](/plugins/webpack/module-federation-plugin-v1).

:::tip
Module Federation v1.0 is no longer being iterated. We recommend using Module Federation v1.5 or v2.0.
:::

### Module Federation v1.5

> This version is natively supported in Rspack, providing features such as module export, module loading, dependency sharing.

Reviewer: ```suggestion
This version is natively supported in Rspack. In addition to supporting module export, module loading, and dependency sharing capabilities of Module Federation v1.0, it also adds runtime plugin functionality, allowing users to extend the behavior and functionality of module federation.
```

You can use it with Rspack's [ModuleFederationPlugin](/plugins/webpack/module-federation-plugin), no need to install any plugins.

```js title="rspack.config.js"
const rspack = require('@rspack/core');

module.exports = {
  output: {
    // set uniqueName explicitly to make HMR work
    uniqueName: 'app',
  },
  plugins: [
    new rspack.container.ModuleFederationPlugin({
      // options
    }),
  ],
};
```

> Refer to: [Module Federation v1.5 Example](https://github.com/rspack-contrib/rspack-examples/tree/main/rspack/module-federation-v1.5).

### Module Federation v2.0

> This is the enhanced version of Module Federation. It provides additional features based on Module Federation v1.5, such as dynamic type hinting, manifest, federation runtime, runtime plugin system, etc. These features make Module Federation more suitable for supporting micro-front-end architecture of large web applications.

Reviewer: ```suggestion
This is the enhanced version of Module Federation. Based on Module Federation v1.5, it provides some additional features out of the box, such as dynamic type hints and manifest, etc. These features make Module Federation more suitable for supporting micro-front-end architecture of large web applications.
```
